common/djangoapps/util/milestones_helpers.py

```diff
@@ -313,6 +314,28 @@ def remove_content_references(content_id):
     return milestones_api.remove_content_references(content_id)
 
 
+def any_unfulfilled_milestones(course_id, user_id):
+    """ Returns a boolean if user has any unfulfilled milestones """
+    if not settings.FEATURES.get('MILESTONES_APP', False):
+        return None
+    from milestones.exceptions import (
+        InvalidMilestoneRelationshipTypeException,
+        InvalidCourseKeyException,
+        InvalidUserException,
+    )
+    try:
+        return bool(
+            get_course_milestones_fulfillment_paths(course_id, {"id": user_id})
+        )
+    # Exception when milestone is not found
+    except InvalidMilestoneRelationshipTypeException:
+        return False
+    except InvalidCourseKeyException:
```

Comment:
This is the question I raised to @BenjiLee -- if it's possible that these exceptions can be tripped by your workflows, you will need to handle for them.  However, if it's not possible for say, an invalid course_id or user_id to be passed to the milestones subsystem, then you can ignore any irrelevant exceptions.